pkg/controllers/grafana-operator/controller.go

```diff
@@ -286,13 +284,35 @@ func (r *reconciler) reconcileSubscription(ctx context.Context) reconcileResult
 		return creationResult(err)
 	}
 
-	r.logger.Info("Updating Grafana Operator Subscription")
-	subscription.Spec.StartingCSV = desired.Spec.StartingCSV
-	return updationResult(r.k8sClient.Update(ctx, &subscription))
+	if subscription.Spec.StartingCSV == desired.Spec.StartingCSV {
+		return next()
+	}
+
+	r.logger.WithValues("Name", subscription.Name).Info("Deleting Subscription")
+	if err := r.k8sClient.Delete(ctx, &subscription); err != nil {
+		return reconcileError(err)
+	}
+
+	r.logger.WithValues("Name", subscription.Status.InstalledCSV).Info("Deleting CSV")
+	csv := v1alpha1.ClusterServiceVersion{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+			Kind:       "ClusterServiceVersion",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      subscription.Status.InstalledCSV,
+			Namespace: Namespace,
+		},
+	}
+	if err := r.k8sClient.Delete(ctx, &csv); err != nil {
+		return reconcileError(err)
+	}
```

Comment:
How about we refactor this to a `r.deleteSubscription()` ?

Reply:
I was thinking to extract an upgrade function instead. I'll submit a follow up PR :+1: